src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveAutowireInlineAttributesPassTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Tests\Compiler;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\DependencyInjection\Compiler\AutowirePass;
+use Symfony\Component\DependencyInjection\Compiler\ResolveAutowireInlineAttributesPass;
+use Symfony\Component\DependencyInjection\Compiler\ResolveChildDefinitionsPass;
+use Symfony\Component\DependencyInjection\Compiler\ResolveClassPass;
+use Symfony\Component\DependencyInjection\Compiler\ResolveNamedArgumentsPass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+require_once __DIR__.'/../Fixtures/includes/autowiring_classes.php';
+
+class ResolveAutowireInlineAttributesPassTest extends TestCase
+{
+    public function testAttribute()
+    {
+        $container = new ContainerBuilder();
+        $container->register(Foo::class)->setAutowired(true);
+
+        $container->register('autowire_inline1', AutowireInlineAttributes1::class)
+            ->setAutowired(true);
+
+        $container->register('autowire_inline2', AutowireInlineAttributes2::class)
+            ->setAutowired(true);
+
+        (new ResolveNamedArgumentsPass())->process($container);
+        (new ResolveClassPass())->process($container);
+        (new ResolveChildDefinitionsPass())->process($container);
```

Comment:
this one should be moved after the new one and we need a test case proving definitions can have a parent

Reply:
I thought since they are executed in a sorted way it may make sense to have them processed in the same order as the code is being executed in reality is this assumption wrong?

Reply:
Besides that I'm still running into test issues there though :disappointed: not sure what I'm missing in general.